src/Sylius/Bundle/ApiBundle/Serializer/ProductVariantNormalizer.php

```diff
@@ -71,9 +72,15 @@ public function normalize($object, $format = null, array $context = [])
             unset($data['originalPrice']);
         }
 
+        /** @var ArrayCollection $appliedPromotions */
         $appliedPromotions = $object->getAppliedPromotionsForChannel($channel);
-        if (!empty($appliedPromotions)) {
-            $data['appliedPromotions'] = $appliedPromotions;
+        if (!$appliedPromotions->isEmpty()) {
+            $data['appliedPromotions'] = array_map(fn (CatalogPromotionInterface $catalogPromotion) => sprintf(
+                '/api/v2/shop/catalog-promotions/%s',
```

Comment:
we talked about it, can we make this path more dynamic? if a user will change prefixes in his project this serialization will cause an error.
if we are not able to make it dynamic, IMO we should move it to the parameter and inject it as an argument to the service and add note to our docs.

Reply:
OR if we do not be able to use IRI here, maybe a better option will be to serialize here necessary data, instead of using IRI.
@lchrusciel @Zales0123 @GSadee, what do you think? which option is best, or maybe you have a better idea?

Reply:
Probably once again, but, what was the origin problem with not using IriConverter here? I've just pulled this branch and checked using IriConverter here and it seems to be working correctly (I've got IRI with shop prefix in shop context)

Reply:
It seems that is actually works well